src/layer/vector/Canvas.js

```diff
@@ -96,11 +97,17 @@ L.Canvas = L.Renderer.extend({
 
 	_draw: function (clear) {
 		this._clear = clear;
-		var layer;
+		var layer, bounds = this._redrawBounds;
+		this._ctx.save();
+		if (bounds) {
+			this._ctx.beginPath();
+			this._ctx.rect(bounds.min.x, bounds.min.y, bounds.max.x - bounds.min.x, bounds.max.y - bounds.min.y);
```

Comment:
The bounds will create artifacts when changing the style of features (e.g. line width on hover). Somehow some padding must be added to these bounds.